swap-protocol-home.js

```diff
@@ -0,0 +1,295 @@
+    default:
+      formattedDenom = denom + '-a';
+      break;
+  }
+  return formattedDenom;
+}
+
+const mapPrices = async (denoms, pricefeedResult) => {
+  // for now drop any of the usd:30 prices returned
+  const nonThirtyPrices = pricefeedResult.filter(p => !p.market_id.includes('30'));
+  let prices = {};
+
+  let mappedPrices = {};
+  for (const price of nonThirtyPrices) {
+    const priceName = price.market_id.split(":")[0];
+    mappedPrices[commonDenomMapper(priceName)] = { price: Number(price.price)};
+
+    // hbtc doesn't have it's own price, it just uses btc's price
+    if (commonDenomMapper(priceName) === 'btcb-a') {
+      mappedPrices['hbtc-a'] = { price: Number(price.price)};
+    }
+  }
+
+  for (const denom of denoms) {
+    let mappedPrice = mappedPrices[denom];
+    let price = { price: 0 };
+
+    if (mappedPrice) {
+      price = { price: mappedPrice.price };
+    }
+    prices[denom] = price;
+  }
+  return prices;
+};
+
+const mapSwpPoolData =  (denoms, siteData, swpPoolDataJson) => {
+  const prices = siteData['prices'];
+  const denomConversions = siteData['denomConversions'];
+
+  const coins = swpPoolDataJson.result.reduce((coinMap, pool) => {
+    const nonUsdxAsset = findNonUsdxTokenInPool(pool);
+    const usdxAsset = findUsdxTokenInPool(pool);
+
+    const formattedNonUsdxDenom = commonDenomMapper(nonUsdxAsset.denom);
+
+    const factor = denomConversions[formattedNonUsdxDenom];
+
+    const nonUsdxAssetValue = Number(nonUsdxAsset.amount) / factor * prices[formattedNonUsdxDenom].price;
+    const usdxAssetValue = Number(usdxAsset.amount) / FACTOR_SIX * prices['usdx'].price
+
+    coinMap[pool.name] = {
+      totalValueLocked: nonUsdxAssetValue + usdxAssetValue,
+      totalShares: Number(pool.total_shares)
+    };
+
+    return coinMap;
+  }, {});
+
+  return coins;
+}
+
+const mapCssIds = (pools) => {
+  let ids = {};
+  ids['TAV'] = 'TAV';
+
+  for (const pool of pools) {
+    ids[pool] = {};
+    ids[pool].totalValueLocked = formatCssId('tvl', pool);
+    ids[pool].rewardApy = formatCssId('rapy', pool);
+  }
+  return ids;
+}
+
+const setTVLAndTAVDisplayValues = async (siteData, cssIds) => {
+  const cssIdTAV = cssIds['TAV'];
+  const totalValueLockedByPool = siteData['swpPoolData'];
+
+  let totalAssetValue = 0;
+  for (const pool in totalValueLockedByPool) {
+    let totalValueLocked = 0;
+    totalValueLocked += totalValueLockedByPool[pool].totalValueLocked;
+
+    const totalValueLockedUsd = usdFormatter.format(totalValueLocked);
+    const cssIdTVL = cssIds[pool].totalValueLocked;
+    setDisplayValueById(cssIdTVL, totalValueLockedUsd);
+
+    totalAssetValue += totalValueLockedByPool[pool].totalValueLocked;
+  }
+
+  const totalAssetValueUsd = usdFormatter.format(totalAssetValue);
+  setDisplayValueById(cssIdTAV, totalAssetValueUsd);
+};
+
+
+const setSwpPrice = async (swpMarketData) => {
+  let swpPriceInUSD = 0;
+  if (swpMarketData) {
+    swpPriceInUSD = swpMarketData.current_price.usd;
+  }
```

Comment:
Is this an adequate empty record to guard against the page bombing out of if Coingecko's API doesn't return what we expect? I tested it out by putting a junk URL and the script still ran, just with a swp price of 0.